web-frontend/src/custom/useMNS.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useState } from 'react';
+
+import { Args, ICallData, MAINNET_CHAIN_ID } from '@massalabs/massa-web3';
+import { bytesToStr } from '@massalabs/web3-utils';
+
+import { usePrepareScCall } from './usePrepareScCall';
+import { contracts, networks } from '@/utils/const';
+
+export function useMNS() {
+  const [targetMnsAddress, setTargetMnsAddress] = useState<string>('');
+  const [mns, setMns] = useState<string>('');
+
+  const { client, account, chainId } = usePrepareScCall();
+
+  const currentNetwork =
+    chainId === MAINNET_CHAIN_ID ? networks.mainnet : networks.buildnet;
+
+  const reverseResolveDns = useCallback(
+    async (address = '') => {
+      const targetAddress = address || account?.address();
+      if (!targetAddress) return;
+      const reverseResolveCallData = {
+        targetAddress: contracts[currentNetwork].mnsContract,
+        targetFunction: 'dnsReverseResolve',
+        parameter: new Args().addString(targetAddress).serialize(),
+      } as ICallData;
+
+      if (!client) return;
+
+      try {
+        const result = await client
+          .smartContracts()
+          .readSmartContract(reverseResolveCallData);
+        setMns(bytesToStr(result.returnValue));
+      } catch (e) {
+        setMns('');
+        console.error(e);
+      }
+    },
+    [client],
+  );
```

Comment:
Probably because of my lack of front end knowledge but there are a few things I don't understand here:
1) Why don't we do an async function like `resolveDns` ?
2) Why is the name of the "returned" var "mns" ? Shouldn't it be something like "domain name" ? If I understood well, "mns" is for "Massa Name Service", so it's weird to name that var like that

Reply:
in reality resolveDns should also be a callback, i was waiting for some review/cleanup changes to implement

Reply:
thank you for pointing this out... indeed we should use the terminology "Massa Domain Name" ? but this is as much a product question that a dev question no @Thykof

Reply:
i was really talking about the variable name, not the name of the service 
It's weird to me that the var is named "mns" while it's supposed to be the list of domain names for a givent address

Reply:
if the variable is a domain name then it must be named `domainName` (or in plural because it's a list)

Reply:
it became ```domainNameList```